### ISSUE TITLE
Add more variables and fix RH calculation in global gird-to-grid verification

### ISSUE THREAD

**Issue by mollybsmith-noaa** [NOAA-GSL/VxIngest#62]
Our global grid-to-grid workflow currently verifies UGRD and VGRD wind, but the global group has requested that I add total wind as well, for comparisons to RAOB verification.

**Comment by mollybsmith-noaa** [comment]
They also want verification of specific humidity and RHobT.

**Comment by mollybsmith-noaa** [comment]
The plan for fixing our RH calculation is to use the Flatau formula to calculate the saturation vapor pressure for a given temperature, and then turn that into a saturation mixing ratio. We will then turn the specific humidity into an actual mixing ratio, and use the ratio of the two mixing ratios to derive RH.

**Comment by mollybsmith-noaa** [comment]
I've outlined the actual coding work that needs to be done for this. Basically, we should only have to alter GFS_init.py, which is the first step in our grid_to_grid verification process. Currently, GFS_init.py identifies the appropriate model files for that hour's verification (on /public or elsewhere), extracts the required variables (HGT, TMP, RH, UWND, VWND), and stores them in a new grib2 file in our working directory. GFS_init.py needs to change to use pygrib to calculate RH from specific humidity and temperature, and then store that calculated field in the working grib2 file (instead of grabbing the existing RH field from the model output). Likewise, it should use pygrib to derive WND from UWND and VWND.

**Comment by mollybsmith-noaa** [comment]
I've added in code to calculate the total wind speed for verification, and am in the process of adding in code to calculate RH (I just need to figure out how to write the derived field to a grib2 file, something that pygrib claims can be done but has absolutely no documentation for). It will take a while to pull five months of data off of the mass store, so in the meantime I'm beginning work on the couchbase MATS app.

**Comment by mollybsmith-noaa** [comment]
The reprocessing is running and is verifying U-wind, V-wind, total wind, temp, specific humidity, relative humidity, and height. All are copied straight from post-processed model files except for total wind and RH, both of which are calculated. Initial tests of the code that does this showed both producing reasonable values, so I have begun the backfilling. There should be at least a month done by Friday, at which point we will have sanity check number 2. One note: for whatever reason, grib2 encoding fails for RH levels at or above 1 mb, but as there is pretty much no moisture up there anyway (RH is everywhere less than a thousandth of a percent), I don't think this is the biggest deal. I'm pretty sure that modelers never look at RH that high up (have asked Dave to confirm). And we are verifying specific humidity up to 0.02 mb.

**Comment by mollybsmith-noaa** [comment]
Because of size limitations on amb-verif's /scratch2 quota, I had to do this reprocecssing in two batches of three months, adding up to the entire six month period. The first tranche of three months is complete, and the results look reasonable, so I am currently loading them into the database (at a glacial pace), where they'll be viewable with METexpress. When that is done I will tip off the second tranche.

**Comment by mollybsmith-noaa** [comment]
Batch two is currently in progress in generating its stat files.

**Comment by mollybsmith-noaa** [comment]
The original backfilling of the global grid-to-grid verification is done, but I am still in the progress of filling the mid- to late-December data gap caused by metv-gsd's log4j shutdown. I am optimistic that everything except the HFIP (which are lower-priority) models (which are lower-priority) will be done and operating in realtime again by the end of the week, and those should be done not long after. I have also included Judy's new L127_dev1 run in our Hera workflow.

**Comment by mollybsmith-noaa** [comment]
This is fully done and running in realtime on Hera, where most of the models are, and almost done on Jet, where L127 and the HFIP models live (it would have been done today if Jet had not been down for maintenance). All that remains on Jet is to schedule a task to load the calculated stats into the database.

**Comment by mollybsmith-noaa** [comment]
This new data is stored in the mv_gsl_global_g2g_rt database, intended to replace the old mv_gsl_global_g2g. I had intended to leave both databases available for a few weeks so that developers could compare, but mv_gsl_global_g2g got corrupted in the holiday shutdown and I've deleted it ahead of schedule. A copy of it exists on model-vxtest if we really need it back (viewable on mats-docker-preint.gsd.esrl.noaa.gov with a VPN).
